src/meta/src/manager/catalog/mod.rs

```diff
@@ -1239,7 +1239,6 @@ where
         let user_core = &mut core.user;
         database_core.ensure_database_id(sink.database_id)?;
         database_core.ensure_schema_id(sink.schema_id)?;
-        database_core.ensure_table_id(sink.associated_table_id)?;
```

Comment:
Should check whether `dependent_relations` exist here.

Reply:
None of the other streaming jobs did this. Maybe do it in a separate pr.

Reply:
All of other streaming jobs do this. You can take a look at `start_create_table_procedure`. Some functions doesn't check `dependent_relations`, because we can find all the dependents without `dependent_relations` variable.

Reply:
Oh I see. Acctually create_index did't do this. Should I also add the check for it?

Reply:
No. Index only depends on its `primary_table_id`, so we just check whether `primary_table_id` exists.

Reply:
Yeah, index's `dependent_relations` only contains `primary_table_id`

Reply:
Them let's add an assert here!